### ISSUE TITLE
Watch for changes.

### ISSUE THREAD

**Issue by jamestalmage** [Benjamin-Dobell/luabundle#5]
Watch the base file, as well as all resolved files for changes, and re-bundle.
Useful for hot-reload scenarios.

**Comment by Benjamin-Dobell** [comment]
I'd probably say this one is a little bit out of scope for this core library; although it's definitely in scope for https://github.com/Benjamin-Dobell/luabundler.

My ideal solution would be a small `luabundle-watch` package that takes a Lua search path and a list of module names. It'd watch for changes in resolved files as well as for the creation of new files that would be resolved instead of the existing resolved files (higher priority in the search path).

Additionally, it'd take the same proposed `resolve` callback as luabundle itself https://github.com/Benjamin-Dobell/luabundle/issues/4

I'll leave this issue open though as a general issue to track any changes to luabundle that may be required to best facilitate a simple integration.

**Comment by jamestalmage** [comment]
Yes. A separate module would be entirely fine. I wouldn't want it bundled into `luabundler` though. I intend to consume this module directly.

I think a low-friction way to solve this would be to provide a lower level API than `bundle` / `bundleString`. Expose a completion callback that consumers takeover from [here](https://github.com/Benjamin-Dobell/luabundle/blob/9210e45d787d76f44132a04b086fa55ed578363d/src/bundle/index.ts#L43).

**Comment by Benjamin-Dobell** [comment]
I'm not yet in dire need of this functionality, and most of me spare time at the moment is taken up by development of [Luanalysis](https://github.com/Benjamin-Dobell/IntelliJ-Luanalysis).

So if you're keen to work on this, then I'll very happily merge PRs that introduce the necessary extension points and/or additional APIs.

**Comment by jamestalmage** [comment]
For sure. Happy to.

Luanalysis is fantastic, BTW.

**Comment by 00fast00** [comment]
I'm using:

```bash
while inotifywait -e close_write src/*.lua; do luabundler bundle src/main.lua --lua=5.1 --path="src/?.lua" --output=dist/gui_overwatch.lua; done
```
